python/apps/taiga/src/taiga/stories/stories/services/__init__.py

```diff
@@ -181,10 +181,26 @@ async def _validate_and_process_values_to_update(story: Story, values: dict[str,
             output["status"] = status
 
             # Calculate new order
-            latest_story = await stories_repositories.list_stories(
-                filters={"status_id": status.id}, order_by=["-order"], offset=0, limit=1
+            output["order"] = await _calculate_next_order(status_id=status.id)
+
+    elif workflow_slug := output.pop("workflow", None):
+        workflow = await workflows_repositories.get_workflow(
+            filters={"project_id": story.project_id, "slug": workflow_slug}, prefetch_related=["statuses"]
+        )
+
+        if not workflow:
+            raise ex.InvalidWorkflowError("The provided workflow is not valid.")
+        elif workflow.slug != story.workflow.slug:
+            output["workflow"] = workflow
+
+            # Set first status
+            first_status = await workflows_repositories.list_workflow_statuses(
+                filters={"workflow_id": workflow.id}, order_by=["order"], offset=0, limit=1
             )
-            output["order"] = DEFAULT_ORDER_OFFSET + (latest_story[0].order if latest_story else 0)
+            output["status"] = first_status[0]
```

Comment:
It'll fail if there's no statuses on the destination workflow

Reply:
Done!